lms/djangoapps/email_marketing/tasks.py

```diff
@@ -294,6 +299,71 @@ def _get_course_content(course_url, sailthru_client, email_config):
     return response
 
 
+def _get_or_create_sailthru_list_for_site(sailthru_client, email_config, site=None, default_list_name=None):
+    """
+    Get the user list name from cache if exists else create one and return the name,
+    callers of this function should perform the enabled check of email config.
+    :param sailthru_client:
+    :param email_config:
+    :param site:
+    :param default_list_name
+    :return: list name if exists or created else return None
+    """
+    if site and site.id != settings.SITE_ID:
+        list_name = site.domain.replace(".", "_") + "_user_list"
+    else:
+        list_name = default_list_name
+
+    _get_or_create_sailthru_list(sailthru_client, email_config, list_name)
+    return list_name
+
+
+def _get_or_create_sailthru_list(sailthru_client, email_config, list_name):
+    """
+    Get list from sailthru and return if list_name exists else create a new one
+    and return list data for all lists.
+    """
+    cached = cache.get(email_config.CACHE_KEY)
+    if cached:
+        for user_list in cached['lists']:
+            if user_list.get('name') == list_name:
+                return cached
+
+    sailthru_get_response = sailthru_client.api_get("list", {})
+    if not sailthru_get_response.is_ok():
+        error = sailthru_get_response.get_error()
+        log.info("Error attempting to read list record from Sailthru: %s", error.get_message())
+        return
+
+    for user_list in sailthru_get_response.json['lists']:
+        if user_list.get('name') == list_name:
+            cache.set(email_config.CACHE_KEY, sailthru_get_response.json)
+            return sailthru_get_response.json
+
+    # create a list in sailthru
+    list_params = {'list': list_name, 'primary': 0, 'public_name': list_name}
+    try:
+        sailthru_response = sailthru_client.api_post("list", list_params)
+    except SailthruClientError as exc:
+        log.error("Exception attempting to list record for key %s in Sailthru - %s", list_name, unicode(exc))
+        return
+
+    if not sailthru_response.is_ok():
+        error = sailthru_response.get_error()
+        log.error("Error attempting to create list in Sailthru: %s", error.get_message())
+        return
+
+    # get all the lists including the new created one
+    sailthru_get_response = sailthru_client.api_get("list", {})
+    if not sailthru_get_response.is_ok():
+        error = sailthru_get_response.get_error()
+        log.info("Error attempting to read list record from Sailthru: %s", error.get_message())
+        return
```

Comment:
Here you can call the new `_get_sailthru_list_map(sailthru_client)` and populate the cache with the result.